### PR TITLE
Form tests cleanup

### LINKED ISSUE #43: Cleanup form_test.go:TestFormValidSlice
This could be a bit more readable: https://github.com/google/go-safeweb/blob/9825ed6ad995cfe447145a5599ffa146d1d1463c/safehttp/form_test.go#L763

### PR DESCRIPTION
**Pull request by mihalimara22**
Fixes #43 

Refactored parts of the form tests to make them less complex and bug-prone.

### CODE REVIEW

**Comment by kele** [comment]
Please squash the commits and give it a meaningful description.

File: safehttp/form_test.go [hunk 1]
```diff
@@ -16,14 +16,13 @@ package safehttp_test
...
 	"errors"
-	"fmt"
 	"math"
 	"net/http"
 	"net/http/httptest"
-	"reflect"
 	"strconv"
 	"strings"
 	"testing"
+	txttemplate "text/template"
```
Comment (kele) [inline safehttp/form_test.go:25]: Do you need the alias?

File: safehttp/form_test.go [hunk 2]
```diff
@@ -1053,37 +925,222 @@ func TestFormErrorHandling(t *testing.T) {
...
+				req := httptest.NewRequest("POST", "/", strings.NewReader("pizza=8&pizza=9&pizza=10"))
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+				return req
+			}(),
+			placeholder: &[]uint64{},
+			want:        &[]uint64{8, 9, 10},
+		},
+		{
+			req: func() *http.Request {
+				req := httptest.NewRequest("POST", "/", strings.NewReader("pizza=margeritta&pizza=diavola&pizza=calzone"))
```
Comment (kele) [inline safehttp/form_test.go:993]: @empijei how do you feel about this? ;p

File: safehttp/form_test.go (hunk 1 above)
Comment (mihalimara22) [reply safehttp/form_test.go]: No, I don't, I removed the alias. I followed the example you sent me because I thought it's the preferred practice.

Comment (kele) [reply safehttp/form_test.go]: Ah, sorry. There was a clash before (I think), or maybe I introduced a clash during development.

In general, we're trying to alias as little as possible. This is also one of the reasons we avoid packages names like `util` or `common`, as people importing them might need to use aliasing a lot.

**Pull request merged**